theurgy.lic

```diff
@@ -479,22 +694,22 @@ class Theurgy
     Flags.add('theurgy-kertigen', 'The thick smell of ozone fills your nostrils', 'The sounds of Kertigen\'s forge still ring in your ears')
     Flags.add('research_done', '^Breakthrough!')
 
-    ######################## SCRIPT ########################
-
+    @actions = TheurgyActions.new(@settings, merged_data)
+    
     pray
-    check_supplies
-    walk_to '741'
-    fput 'commune sense'
-    waitrt?
-
-    ritual_order = %i[tithe study_wall reflectory carve_bead find_seed enter_altar bathe meditate_bead pray_badge dance incense_wine recite_prayer clean_altar pray exit_altar invoke_parchment]
-    ritual_order.each do |ritual|
-      commune_check
-      send(ritual)
-    end
+    @actions.buy_supplies
+    @actions.perform_next_action while @actions.rituals_remain? && should_continue?
 
     fput 'commune'
   end
+
+  def should_continue?
+    if (@settings.theurgy_exp_threshold || 0) > 0
```

Comment:
Doesn't this always evaluate to true? If threshold exists or 0 is greater than 0?